### ISSUE TITLE
unable to install the jupyter extension tutorial M1 mac

### ISSUE THREAD

**Issue by johndavidstl** [jupyterlab/jupyterlab#15229]
## Description

I'm going through the[ Jupyter Extension Tutorial](https://jupyterlab.readthedocs.io/en/stable/extension/extension_tutorial.html#extension-tutorial) and I am unable to complete the following command:

pip install -ve .

I get an error that starts at the resolution step. I have tried the steps from previous reports and have cleared the npm cache. 

` ➤ YN0000: ┌ Resolution step
  ➤ YN0001: │ RequestError: tunneling socket could not be established, cause=connect ECONNREFUSED 1.2.3.4:1234
      at ClientRequest.<anonymous> (/private/var/folders/yb/67fjc235007gxm_f4vr4wjxw0000gq/T/pip-build-env-votky4jv/overlay/lib/python3.11/site-packages/jupyterlab/staging/yarn.js:195:14340)
      at Object.onceWrapper (node:events:629:26)
      at ClientRequest.emit (node:events:526:35)
      at o.emit (/private/var/folders/yb/67fjc235007gxm_f4vr4wjxw0000gq/T/pip-build-env-votky4jv/overlay/lib/python3.11/site-packages/jupyterlab/staging/yarn.js:190:90286)
      at ClientRequest.u (/private/var/folders/yb/67fjc235007gxm_f4vr4wjxw0000gq/T/pip-build-env-votky4jv/overlay/lib/python3.11/site-packages/jupyterlab/staging/yarn.js:196:131)
      at Object.onceWrapper (node:events:629:26)
      at ClientRequest.emit (node:events:514:28)
      at Socket.socketErrorListener (node:_http_client:501:9)
      at Socket.emit (node:events:514:28)
      at emitErrorNT (node:internal/streams/destroy:151:8)
      at ClientRequest.u (/private/var/folders/yb/67fjc235007gxm_f4vr4wjxw0000gq/T/pip-build-env-votky4jv/overlay/lib/python3.11/site-packages/jupyterlab/staging/yarn.js:196:28)
      at Object.onceWrapper (node:events:629:26)
      at ClientRequest.emit (node:events:514:28)
      at Socket.socketErrorListener (node:_http_client:501:9)
      at Socket.emit (node:events:514:28)
      at emitErrorNT (node:internal/streams/destroy:151:8)
      at emitErrorCloseNT (node:internal/streams/destroy:116:3)
      at process.processTicksAndRejections (node:internal/process/task_queues:82:21)
  ➤ YN0000: └ Completed in 39s 404ms
  ➤ YN0000: Failed with errors in 39s 405ms
  Traceback (most recent call last):
    File "/Users/DFZ229/miniconda3/envs/jupyterlab-ext/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 167, in prepare_metadata_for_build_editable
      hook = backend.prepare_metadata_for_build_editable
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  AttributeError: module 'hatchling.build' has no attribute 'prepare_metadata_for_build_editable'

  During handling of the above exception, another exception occurred:

  Traceback (most recent call last):
    File "/Users/DFZ229/miniconda3/envs/jupyterlab-ext/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
      main()
    File "/Users/DFZ229/miniconda3/envs/jupyterlab-ext/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
      json_out['return_val'] = hook(**hook_input['kwargs'])
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "/Users/DFZ229/miniconda3/envs/jupyterlab-ext/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 176, in prepare_metadata_for_build_editable
      whl_basename = build_hook(metadata_directory, config_settings)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "/private/var/folders/yb/67fjc235007gxm_f4vr4wjxw0000gq/T/pip-build-env-votky4jv/overlay/lib/python3.11/site-packages/hatchling/build.py", line 78, in build_editable
      return os.path.basename(next(builder.build(wheel_directory, ['editable'])))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "/private/var/folders/yb/67fjc235007gxm_f4vr4wjxw0000gq/T/pip-build-env-votky4jv/overlay/lib/python3.11/site-packages/hatchling/builders/plugin/interface.py", line 150, in build
      build_hook.initialize(version, build_data)
    File "/private/var/folders/yb/67fjc235007gxm_f4vr4wjxw0000gq/T/pip-build-env-votky4jv/normal/lib/python3.11/site-packages/hatch_jupyter_builder/plugin.py", line 87, in initialize
      raise e
    File "/private/var/folders/yb/67fjc235007gxm_f4vr4wjxw0000gq/T/pip-build-env-votky4jv/normal/lib/python3.11/site-packages/hatch_jupyter_builder/plugin.py", line 82, in initialize
      build_func(self.target_name, version, **build_kwargs)
    File "/private/var/folders/yb/67fjc235007gxm_f4vr4wjxw0000gq/T/pip-build-env-votky4jv/normal/lib/python3.11/site-packages/hatch_jupyter_builder/utils.py", line 113, in npm_builder
      run([*npm_cmd, "install"], cwd=str(abs_path))
    File "/private/var/folders/yb/67fjc235007gxm_f4vr4wjxw0000gq/T/pip-build-env-votky4jv/normal/lib/python3.11/site-packages/hatch_jupyter_builder/utils.py", line 229, in run
      return subprocess.check_call(cmd, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "/Users/DFZ229/miniconda3/envs/jupyterlab-ext/lib/python3.11/subprocess.py", line 413, in check_call
      raise CalledProcessError(retcode, cmd)
  subprocess.CalledProcessError: Command '['/private/var/folders/yb/67fjc235007gxm_f4vr4wjxw0000gq/T/pip-build-env-votky4jv/overlay/bin/jlpm', 'install']' returned non-zero exit status 1.
  error: subprocess-exited-with-error
  
  × Preparing editable metadata (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> See above for output.
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  full command: /Users/DFZ229/miniconda3/envs/jupyterlab-ext/bin/python3.11 /Users/DFZ229/miniconda3/envs/jupyterlab-ext/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py prepare_metadata_for_build_editable /var/folders/yb/67fjc235007gxm_f4vr4wjxw0000gq/T/tmprrm4lf50
  cwd: /Users/DFZ229/EFP/jupyter-extension-testing/my_first_extension
  Preparing editable metadata (pyproject.toml) ... error
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
`

## Reproduce

Use an M1 Mac and follow the Intro Jupyter Extension tutorial. 

## Expected behavior

On my personal laptop, which is an 2018 Macbook Pro (x86), the build is successful. 

## Context

- Operating System and version: M1 ARM Macbook Pro
- Browser and version: Chrome
- JupyterLab version: latest

**Comment by krassowski** [comment]
This appears to be an issue with mis-configured yarn proxy (unrelated to jupyterlab itself). Did you try the suggestions from https://github.com/yarnpkg/yarn/issues/5599? Alternatively either:
- https://stackoverflow.com/questions/33162560/error-tunneling-socket-while-executing-npm-install or
- https://stackoverflow.com/questions/72457804/yarn-installation-error-tunneling-socket-could-not-be-established-cause-conne
could have some useful ideas but both the former is npm-centric whereas JupyterLab uses yarn.

**Comment by johndavidstl** [comment]
I believe I have the yarn and npm proxy setting correctly configured for my environment. I am able to run 'yarn install' and 'npm install' successfully.

**Comment by krassowski** [comment]
By running `yarn install` successfully, do you mean in the same environment and in `jupyterlab`? Which version of yarn? Does it also work with `jlpm install`? I have a suspicion that this could be version dependent (on latest we use berry) and I wonder if the suggestion from https://github.com/yarnpkg/yarn/issues/5599#issuecomment-1491457079 about `HTTP_PROXY`/`HTTPS_PROXY` could help you.

**Comment by johndavidstl** [comment]
I was able to finally run` pip install -ve .`

The key to it was adding the correct setting inside the .yarnrc.yml file:

npmRegistryServer: "https://xxxxxinternalfacingnpm..."

This solution is very specific to proxy set up I have to work with, but it definitely allows the build step to complete. 

I do have another issue now as I'm working through the extension tutorial after making the first modification to the JupyterFrontEndPlugin: 

jlab_core.be6103fe6f6cc2c18378.js?v=be6103fe6f6cc2c18378:1 Language pack 'en_US' not installed!
fetch @ jlab_core.be6103fe6f6cc2c18378.js?v=be6103fe6f6cc2c18378:1
await in fetch (async)
activate @ jlab_core.be6103fe6f6cc2c18378.js?v=be6103fe6f6cc2c18378:1
await in activate (async)
(anonymous) @ jlab_core.be6103fe6f6cc2c18378.js?v=be6103fe6f6cc2c18378:1
Promise.then (async)
activatePlugin @ jlab_core.be6103fe6f6cc2c18378.js?v=be6103fe6f6cc2c18378:1
resolveOptionalService @ jlab_core.be6103fe6f6cc2c18378.js?v=be6103fe6f6cc2c18378:1
(anonymous) @ jlab_core.be6103fe6f6cc2c18378.js?v=be6103fe6f6cc2c18378:1
activatePlugin @ jlab_core.be6103fe6f6cc2c18378.js?v=be6103fe6f6cc2c18378:1
resolveOptionalService @ jlab_core.be6103fe6f6cc2c18378.js?v=be6103fe6f6cc2c18378:1
(anonymous) @ jlab_core.be6103fe6f6cc2c18378.js?v=be6103fe6f6cc2c18378:1
activatePlugin @ jlab_core.be6103fe6f6cc2c18378.js?v=be6103fe6f6cc2c18378:1
resolveRequiredService @ jlab_core.be6103fe6f6cc2c18378.js?v=be6103fe6f6cc2c18378:1
(anonymous) @ jlab_core.be6103fe6f6cc2c18378.js?v=be6103fe6f6cc2c18378:1
activatePlugin @ jlab_core.be6103fe6f6cc2c18378.js?v=be6103fe6f6cc2c18378:1
(anonymous) @ jlab_core.be6103fe6f6cc2c18378.js?v=be6103fe6f6cc2c18378:1
start @ jlab_core.be6103fe6f6cc2c18378.js?v=be6103fe6f6cc2c18378:1
K @ 812.93b4681c78d38d76145b.js?v=93b4681c78d38d76145b:1
load (async)
e @ main.df4a154f94063e34ef18.js?v=df4a154f94063e34ef18:1
await in e (async)
37559 @ main.df4a154f94063e34ef18.js?v=df4a154f94063e34ef18:1
t @ main.df4a154f94063e34ef18.js?v=df4a154f94063e34ef18:1
(anonymous) @ main.df4a154f94063e34ef18.js?v=df4a154f94063e34ef18:1
(anonymous) @ main.df4a154f94063e34ef18.js?v=df4a154f94063e34ef18:1

I tried to conda install the language pack, but the error is still there.
